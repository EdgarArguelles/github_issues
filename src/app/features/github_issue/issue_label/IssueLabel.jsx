import './IssueLabel.scss';
import React from 'react';
import PropTypes from 'prop-types';

const IssueLabel = ({label}) => {
    const borderColor = `#${label.color}`;
    const backgroundColor = `#${label.color}26`;

    return <span className="issue-label" style={{borderColor, backgroundColor}}>{label.name}</span>;
};

IssueLabel.propTypes = {
    label: PropTypes.object.isRequired,
};

export default React.memo(IssueLabel);