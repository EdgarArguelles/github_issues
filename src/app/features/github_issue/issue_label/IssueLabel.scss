.issue-label {
  display: inline-block;
  border: 2px solid;
  border-radius: 2em;
  font-size: 0.8rem;
  padding: 0 7px;
  margin: 0 5px;
}