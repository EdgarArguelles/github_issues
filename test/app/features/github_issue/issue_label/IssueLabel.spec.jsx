import renderer from 'react-test-renderer';
import React from 'react';
import IssueLabel from 'app/features/github_issue/issue_label/IssueLabel';

describe('Features -> GithubIssue -> IssueLabel', () => {
    let component, wrapper;

    const mountComponent = props => {
        wrapper = renderer.create(<IssueLabel {...props}/>);
        component = wrapper.root.find(el => el.type.name === 'IssueLabel');
    };

    describe('load', () => {
        it('should have all props', () => {
            mountComponent({
                label: {color: 'red', name: 'test'},
            });

            expect(component.props.label).toStrictEqual({color: 'red', name: 'test'});
        });
    });

    describe('snapshot', () => {
        it('should render IssueLabel with props', () => {
            mountComponent({
                label: {color: 'red', name: 'test'},
            });

            expect(wrapper.toJSON()).toMatchSnapshot();
        });
    });
});