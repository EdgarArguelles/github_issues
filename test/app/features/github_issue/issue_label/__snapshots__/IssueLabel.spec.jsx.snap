// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features -> GithubIssue -> IssueLabel snapshot should render IssueLabel with props 1`] = `
<span
  className="issue-label"
  style={
    Object {
      "backgroundColor": "#red26",
      "borderColor": "#red",
    }
  }
>
  test
</span>
`;
